lib/fizzy/execute.cpp

```diff
@@ -323,6 +326,26 @@ execution_result execute(Instance& instance, FuncIdx function, std::vector<uint6
                 stack.push(ret.stack[0]);
             break;
         }
+        case Instr::return_:
+        case_return:
+        {
+            // TODO: Not needed, but satisfies the assert in the end of the main loop.
+            labels.clear();
+
+            const bool have_result =
+                !instance.module.typesec[instance.module.funcsec[function]].outputs.empty();
```

Comment:
Add an assert for `instance.module.typesec.size() >= instance.module.funcsec[function]`. At least we did that so far, even though the indexing should also throw.